PDFReportEnergie/module.php

```diff
@@ -0,0 +1,250 @@
+        $valueText = $this->Translate('Actual usage:') . ' ' . $data[1];
+        $consumText = $this->Translate("You could $data[6] your consumption by %s in the period");
+        $consumText2 = sprintf($consumText, $data[5]);
+
+        $text =
+        <<<EOT
+        <p> $consum </p>
+        </br></br>
+        <h2 style="font-weight: normal; font-size: 25px">$title</h2>
+        <hr style="height: 5px">
+        <br> </br>
+        <h3>$data[0]</h3>
+            <p>
+            $data[3] <br>
+            $predictionText <br>
+            $valueText <br/>
+            <br>
+            $consumText2<br/>
+            </p>
+        EOT;
+
+        if ($data[7] != -1) {
+            $text .= "<h3>CO2 Emmisionen</h3> <p>$data[7] kg</p>";
+        }
+        return $text;
+    }
+
+    private function FetchData()
+    {
+        $archivID = IPS_GetInstanceListByModuleID('{43192F0B-135B-4CE7-A0A7-1475603F3060}')[0];
+        $startTime = strtotime('first day of last month 00:00:00');
+        $endTime = strtotime('first day of this month 00:00:00');
+
+        $counterID = $this->ReadPropertyInteger('CounterID');
+
+        $month = $this->Translate(date('F', $startTime));
+
+        $consum = AC_GetAggregatedValues($archivID, $counterID, 3, $startTime, $endTime, 0)[0]['Avg'];
+        $consumLastYear = AC_GetAggregatedValues($archivID, $counterID, 3, strtotime('-1 year', $startTime), strtotime('-1 year', $endTime), 0)[0]['Avg'];
+
+        if (($temperatureID = $this->ReadPropertyInteger('TemperatureID')) != 0) {
+            $avgTemp = AC_GetAggregatedValues($archivID, $temperatureID, 3, $startTime, $endTime, 0)[0]['Avg'];
+            $avgTemp = round($avgTemp, 1);
+            $avgTemp = $this->Translate('Die Durchschnittstemperatur betrug: ') . GetValueFormattedEx($temperatureID, $avgTemp);
+        } else {
+            $avgTemp = '';
+        }
+
+        $predictionID = $this->ReadPropertyInteger('PredictionID');
+        $prediction = GetValue($predictionID);
+
+        $percent = round(((1 - ($consum / $prediction)) * 100), 2);
+
+        if ($percent <= 100) {
+            $percentText = $this->Translate('redruce');
+        } else {
+            $percentText = $this->Translate('raise');
+        }
+
+        if ($this->ReadPropertyInteger('CO2Type') != -1) {
+            $co2 = $consum * $this->ReadPropertyInteger('CO2Type');
+        } else {
+            $co2 = -1;
+        }
+
+        //Formatted Values
+        if ($consumLastYear != 0) {
+            $consumLastYear = sprintf($this->Translate('Im Vorjahr hatten Sie einen Verbrauch von %s.'), GetValueFormattedEx($counterID, $consumLastYear));
+        } else {
+            $consumLastYear = '';
+        }
+        $prediction = GetValueFormattedEx($predictionID, $prediction);
+        $consum = GetValueFormattedEx($counterID, $consum);
+        $percent = $percent . '%';
+
+        $data = [$month, $consum, $consumLastYear, $avgTemp, $prediction, $percent, $percentText, $co2];
```

Comment:
Hatten wir dies in anderen Modul auch so gemacht?

Der Nachteil ist, das du immer Indizes hast, welche nichts aussagen. Schöner und lesbarer wäre folgendes:

```
$data = [
  "month" => $month,
  "consumption" => $consumption,
  ....
]
```